helpers/__tests__/request-test.ts

```diff
@@ -95,7 +95,7 @@ describe('parseError', () => {
     ).resolves.toBe('Error1. Error2');
   });
 
-  it('should parse error and return default message', () => {
+  it('should parse error object and return default message', () => {
```

Comment:
Not related to the PR: this test is incredibly difficult to understand IMO. Also, the loop only runs 2 times, so I'm not sure this really does test anything because the call after only is the third one. WDYT ?

Reply:
Yeah I agree it's not the best way of testing promises. I could replace everything with some async/await.

Reply:
I'm not sure which loop you are talking about ? It's in a different test right ?

Reply:
The loop line 217. My understanding is that we want to leave room for more than 3 calls and check that only 3 calls happens. But I feel like we only leave room for 3 calls (2 in the loop, one after) and therefore we're not testing the edge case ('should stop after 3 calls').

Reply:
Indeed, I fixed this by adding one more check at the end that it's not called anymore